wgengine/netstack/netstack.go

```diff
@@ -156,50 +186,91 @@ func (ns *Impl) updateDNS(nm *netmap.NetworkMap) {
 	ns.dns = DNSMapFromNetworkMap(nm)
 }
 
+func (ns *Impl) addSubnetAddress(pn tcpip.NetworkProtocolNumber, addr tcpip.Address) (ok bool) {
+	ip, ok := netaddr.FromStdIP(net.IP(addr))
+	if !ok {
+		return false
+	}
+	ns.mu.Lock()
+	ns.connsOpenBySubnetIP[ip]++
+	ns.mu.Unlock()
+	ns.ipstack.AddAddress(nicID, pn, addr)
+	return true
+}
+
+func (ns *Impl) removeSubnetAddress(addr tcpip.Address) (ok bool) {
```

Comment:
I'd make both addSubnetAddress and removeSubnetAddress take a netaddr.IP instead. You already have it at the caller.

Also, why do these return a bool? It's not used, and doesn't mean much. The caller can abort before calling these if it can't get a netaddr.IP.